src/System.Private.CoreLib/shared/System/Collections/Generic/Dictionary.cs

```diff
@@ -1402,14 +1405,14 @@ public struct Enumerator : IEnumerator<TKey>, IEnumerator
                 private readonly Dictionary<TKey, TValue> _dictionary;
                 private int _index;
                 private readonly int _version;
-                private TKey _currentKey;
+                [AllowNull,MaybeNull] private TKey _currentKey;
```

Comment:
Nit: space after `,`

Reply:
What is the purpose of having the `MaybeNull` here?

Reply:
> Nit: space after ,

Fixed.

> What is the purpose of having the MaybeNull here?

If TKey is a reference type, then prior to MoveNext being called (or after Reset is called), _currentKey will be null.